src/vaults/cross-chain-vault/escrow/base/L2BridgeEscrowBase.sol

```diff
@@ -0,0 +1,109 @@
+//SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.16;
+
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {WETH as IWETH} from "solmate/src/tokens/WETH.sol";
+import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+
+import {BridgeEscrow} from "../BridgeEscrow.sol";
+import {L2Vault} from "src/vaults/cross-chain-vault/L2Vault.sol";
+import {IAcrossBridge} from "src/interfaces/IAcrossBridge.sol";
+
+
+contract L2BridgeEscrowBase is BridgeEscrow {
+    using SafeTransferLib for ERC20;
+    using SafeTransferLib for IWETH;
+
+    /// @notice The L2Vault.
+    L2Vault public vault;
+
+    IAcrossBridge public acrossBridge;
+    address public l1EscrowAddress;
+    IWETH public constant WETH = IWETH(payable(0x4200000000000000000000000000000000000006));
+
+    constructor(L2Vault _vault) BridgeEscrow(_vault) {
+        vault = _vault;
+        asset.safeApprove(address(acrossBridge), type(uint256).max);
+        acrossBridge = IAcrossBridge(payable(0x09aea4b2242abC8bb4BB78D537A67a245A7bEC64));
+    }
+
+    fallback() external payable {}
+    
+    receive() external payable {}
+
+    modifier onlyGovernance() {
+        require(msg.sender == governance, "Only Governance.");
+        _;
+    }
+
+    // function to withdraw eth from the contract
+    function withdrawEth(uint256 _amount) external onlyGovernance {
+        payable(governance).transfer(_amount);
+    }
+
+    // function to withdraw tokens from the contract
+    function withdrawToken(address _token, uint256 _amount) external onlyGovernance {
+        ERC20(_token).safeTransfer(governance, _amount);
+    }
+
+    function setAcrossBridge(address _acrossBridgeAddress) external onlyGovernance {
```

Comment:
Should have a check to validate if this is a valid bridge contract.

Reply:
Is there even a way to make sure? I can try to see if some of the read variables in the ABI return something but those can still be spoofed with a similar abi